src/main/scala/com/microsoft/azure/cosmosdb/spark/CosmosDBConnection.scala

```diff
@@ -391,6 +391,11 @@ private[spark] case class CosmosDBConnection(config: Config) extends CosmosDBLog
       Tuple2.apply(cfDocuments.iterator(), nextContinuation)
     } else 
     {
+      // next Continuation Token is plain and simple when not using Streaming because
+      // all records will be processed. The parameter 'maxPagesPerBatch' is irrelevant
+      // in this case - so there doesn't need to be any suffix in the continutaion token returned
+      nextContinuation = feedResponse.getResponseContinuation()
```

Comment:
Is that because when using batch with useChangeFeed = true, it's expected from the user to load everything till the end of the change feed in the batch process while in streaming, you just need one block ?

Reply:
This only becomes an issue when the changefeed stream is getting persisted (for both streaming and batching). For streaming the customer expectation that each batch/iteration has a reaosnably small size is more common and understandable. When a customer uses batch mode and is trying to persist the entire DataFrame I think it is easy to explain why that isn't going to work. Makes sense?

Reply:
yeah, that makes sense to me